### PR TITLE
Removes `@emotion/react` is not included in builds

### PR DESCRIPTION
**Pull request by TheSonOfThomp**
## ✍️ Proposed changes

Our packages will not build if `@leafygreen/emotion@4.0.2` or greater is installed. This is because this package listed `@emotion/react` as a dependency, which was also an implicit dependency of all other packages.

This packages updates the rollup & babel configs to remove this implicit dependency.

🎟 _Jira ticket:_ [Name of ticket](https://jira.mongodb.org/browse/[name-of-ticket])

## 🛠 Types of changes

- [ ] Tooling (updates to workspace config or internal tooling)
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Style change (purely visual updates; if this is the case, attach a screenshot below!)

## ✅ Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have run `yarn changeset` and documented my changes

- [ ] I have added my new package to the global tsconfig
- [ ] I have added my new package to the Table of Contents on the global README

## Feedback Levels

Please try to mark your comments with one of these levels

1. Praise (👏 🚀 🔥 🍾)
2. Opinions, Ideas, Suggestions (💡 👀)
3. Let’s discuss @ Code review (🤷 🤔 🧐 )
4. Doesn’t meet agreed upon standards (😬 ⚠️ ‼️)
5. Missing an edge case (🛑 🙅 ❌)
6. Missing core functionality (⛔️ ☣️ 🚧 🌋 💀)

(Note: Use ⌘-⌃-␣ to open the Emoji keyboard!)

### CODE REVIEW

File: .changeset/happy-hounds-hope.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+'@leafygreen-ui/text-area': patch
+'@leafygreen-ui/text-input': patch
+'@leafygreen-ui/toast': patch
+'@leafygreen-ui/toggle': patch
+'@leafygreen-ui/tokens': patch
+'@leafygreen-ui/tooltip': patch
+'@leafygreen-ui/typography': patch
+---
+
+- Updates the build to remove an implicit dependency on `@emotion/react`. LG packages will not build if `@leafygreen/emotion@4.0.2` or greater is installed since this package has since removed `@emotion/react` as a dependency.
```
Comment (DesignerDave) [inline .changeset/happy-hounds-hope.md:51]: Can we update this wording to not sound like a breaking change? It sounds like users of the library can't use 4.0.2 or greater with this change.

A good fix would be to say:
"Fixes an issue where LeafyGreen packages would not build if `@leafygreen/emotion@4.0.2` or greater is installed by removing an implicit build dependency on `@emotion/react`.

**Review by DesignerDave — APPROVED** [review]
Let's just double check the file size action results before merging to make sure that this change didn't cause more code to be bundled.

Otherwise, everything's building fine. LGTM!

**Comment by TheSonOfThomp** [comment]
> Let's just double check the file size action results before merging to make sure that this change didn't cause more code to be bundled.
> 
> Otherwise, everything's building fine. LGTM!

It's a little hard to parse but everything looks fine there. Mostly just the icons

**Comment by DesignerDave** [comment]
Yep, looks good to go.
